### ISSUE TITLE
node-gyp rebuild failed for @nodert-win10-cu\windows.system.diagnostics

### ISSUE THREAD

**Issue by ritazh** [NodeRT/NodeRT#83]
`npm i @nodert-win10-cu/windows.system.diagnostics` ends in this error:

```
@nodert-win10-cu\windows.system.diagnostics\build\binding.vcxproj(52,306): error MSB4025: The project file could not be loaded. ', hexadecimal value 0x08
, is an invalid character. Line 52, position 306.
```
Issue seems to be related to this line in `binding.gyp` causing `node-gyp rebuild --msvs_version=2015` to fail:
```
'AdditionalUsingDirectories' : [
     'C:\Program Files (x86)\Windows Kits\10\UnionMetadata\10.0.15063.0'
 ]
```
After updating this file, now getting this error:
```
'SystemMemoryUsage': is not a member of 'Windows::System::Diagnostics'
```
**Environment:**
- Windows OS 1703, build 10.0.15063.483
- node v6.11.2, npm v5.2.0, node-gyp v3.6.2

cc @felixrieseberg @nadavbar

**Comment by bnookala** [comment]
It looks like the SDK had some layout changes, and the correct `Windows.Winmd` file is no longer at `C:\Program Files(x86)\Windows Kits\10\UnionMetadata` but rather at `C:\Program Files(x86)\Windows Kits\10\UnionMetadata\10.0.15063.483\Windows.winmd`

**Comment by felixrieseberg** [comment]
Ohhhh nice sleuthing @bhargav. That seems fixable.

**Comment by bnookala** [comment]
I fixed the vcxproj build issue (it was caused by the usage of a File path instead of a URI). The build issues, however, are beyond me 🤕

**Comment by nadavbar** [comment]
@ritazh  What is the current status? Are you still seeing issues after @bnookala fix?

**Comment by ritazh** [comment]
Hi @nadavbar, yes. that PR only fixes the invalid character issue, which is equivalent to fixing it by hand, which results in a different error, as you can see from above.

**Comment by nadavbar** [comment]
My educated guess here is that these APIs are only supported for apps that are developed with VS 2017. There's a couple of open issues about adding VS 2017 support..shouldn't be too hard to do if node-gyp already supports VS 2017.

**Comment by ritazh** [comment]
Thanks @nadavbar Looks like node-gyp 3.6.2 supports VS2017, which is why I upgraded it and downloaded VS2017, but still getting errors. Happy to help fix this, but not sure where to look next.

**Comment by nadavbar** [comment]
What build flag are you using? If it's still with msvs_version=2015 the build will default to use the VS 2015 compiler.

**Comment by ritazh** [comment]
I used msvs_version=2017

**Comment by nadavbar** [comment]
Hmmm...not sure then. Assuming the configuration is OK and the same Winmd is used for generation and build it might be some specific case in the API that we are missing.

**Comment by hairui219** [comment]
I was able to fix this problem for Creator Update libraries. Code generated and built with vs2017 successfully.

The idea is to update the ```\src\NodeRTLib\ProjectTemplates\binding.gyp``` file as below 

1. add Creator Update version of ```windows.winmd``` file (located at ```%ProgramFiles%/Windows Kits/10/UnionMetadata/10.0.15063.0```) before the normal ```windows.winmd``` file located at ```%ProgramFiles%/Windows Kits/10/UnionMetadata```. Hope this fix could be compatible with older version of Windows 10. **Not tested yet.** 

2. The original code ```'{AdditionalWinmdPath}'``` causes build error as described in this issue. So I just removed it and replaced it with hard coded paths.

So the code would be like:

```
['WIN_VER=="v10"', {
          'msvs_settings': {
            'VCCLCompilerTool': {
              'AdditionalUsingDirectories' : [
                '%ProgramFiles(x86)%/Microsoft Visual Studio 14.0/VC/lib/store/references',
                '%ProgramFiles%/Windows Kits/10/UnionMetadata/10.0.15063.0',
                '%ProgramFiles%(x86)/Windows Kits/10/UnionMetadata/10.0.15063.0'
		'%ProgramFiles%/Windows Kits/10/UnionMetadata',
		'%ProgramFiles(x86)%/Windows Kits/10/UnionMetadata',
		'%ProgramFiles%/Microsoft Visual Studio 14.0/VC/lib/store/references'
		]
              }
            }
          }],
          ['USE_ADDITIONAL_WINMD=="true"', {
          'msvs_settings': {
            'VCCLCompilerTool': {
              'AdditionalUsingDirectories' : [
                '%ProgramFiles%/Windows Kits/10/UnionMetadata/10.0.15063.0',
                '%ProgramFiles%(x86)/Windows Kits/10/UnionMetadata/10.0.15063.0'
		'%ProgramFiles%/Windows Kits/10/UnionMetadata',
		'%ProgramFiles(x86)%/Windows Kits/10/UnionMetadata',
                ]
               }
             }
          }]
```

**Comment by Skriglitz** [comment]
I'm running into this same issue. Upon digging it looks like the path is being autogenerated with a single `\` which when parsed by node-gyp is fine until it's followed by hex which in this case it comes out to `\10` and then is treated as a non-displayable character.  I ran into this issue when pulling from npm or using either the cli or UI generators. If I correct it to `/` or use `\\` it works perfectly fine however.

**Comment by MarkTiedemann** [comment]
I think this was fixed by [#101](https://github.com/NodeRT/NodeRT/pull/101).

**Comment by nadavbar** [comment]
Thanks @MarkTiedemann, closing this issue
